src/main/groovy/com/bmuschko/gradle/docker/tasks/image/DockerSaveImage.groovy

```diff
@@ -36,17 +51,42 @@ class DockerSaveImage extends AbstractDockerRemoteApiTask {
 
     DockerSaveImage() {
         useCompression.set(false)
+        onlyIf {
+            images.getOrNull() || image.present
+        }
+    }
+
+    // part of work-around for https://github.com/docker-java/docker-java/issues/1872
+    @CompileDynamic
+    SaveImagesCmd.Exec getExecution() {
+        dockerClient.saveImagesCmd().@execution
     }
 
     @Override
     void runRemoteCommand() {
-        SaveImageCmd saveImageCmd = dockerClient.saveImageCmd(image.get())
-        InputStream image = saveImageCmd.exec()
+        Set<String> images = images.getOrElse([] as Set)
+        if (image.present) {
+            images = images + image.get()
+        }
+        // part of work-around for https://github.com/docker-java/docker-java/issues/1872
+        SaveImagesCmd saveImagesCmd = new SaveImagesCmdImpl(execution) {
+            @Override
+            List<TaggedImage> getImages() {
+                images.collect {
```

Comment:
Would it work to use `images as List<TaggedImage>` instead?

Reply:
No:
```
class java.lang.String cannot be cast to class com.github.dockerjava.api.command.SaveImagesCmd$TaggedImage (java.lang.String is in module java.base of loader 'bootstrap'; com.github.dockerjava.api.command.SaveImagesCmd$TaggedImage is in unnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @3b15ef58)
```